resources/assets/coffee/_classes/store-supporter-tag.coffee

```diff
@@ -0,0 +1,140 @@
+    @updateCostDisplay()
+
+  initializeSlider: =>
+    $(@slider).slider
+      range: 'min'
+      value: @sliderValue(@MIN_VALUE)
+      min: @sliderValue(@MIN_VALUE)
+      max: @sliderValue(@MAX_VALUE)
+      step: 1
+      animate: true
+      slide: @onSliderValueChanged
+      change: @onSliderValueChanged
+
+  initializeSliderPresets: =>
+    $(@sliderPresets).on 'click', (event) =>
+      target = event.currentTarget
+      price = StoreSupporterTagPrice.durationToPrice(target.dataset.months)
+      $(@slider).slider('value', @sliderValue(price)) if price
+
+  initializeUsernameInput: =>
+    $(@usernameInput).on 'input', @onInput
+
+  getUser: (username) =>
+    $.post laroute.route('users.check-username-exists'), username: username
+    .done (data) =>
+      # make a User DTO?
+      @user =
+        userId: data.user_id
+        username: data.username
+        avatarUrl: data.avatar_url
+
+    .fail (xhr) =>
+      @user = null
+      if xhr.status == 401
+        osu.popup osu.trans('errors.logged_out'), 'danger'
+
+    .always =>
+      @searching = false
+      @updateCart(@user)
+      @updateSearchResult()
+
+  calculate: (position) =>
+    new StoreSupporterTagPrice(Math.floor(position / @RESOLUTION))
+
+  onSliderValueChanged: (event, ui) =>
+    @cost = @calculate(ui.value)
+    @updateCostDisplay()
+
+  onInput: (event) =>
+    if !@searching
+      @searching = true
+      @updateSearchResult()
+    @debouncedGetUser(event.currentTarget.value)
+
+  setUserInteraction: (enabled) =>
+    $(@el).toggleClass('store-supporter-tag--disabled', !enabled)
+    $('.js-slider').slider('disabled': !enabled)
+
+  sliderValue: (price) ->
+    price * @RESOLUTION
+
+  updateCart: (data) ->
+    StoreCart.setEnabled(data?)
+
+  updateSearchResult: =>
+    if @searching
+      @inputFeedback.textContent = Lang.get('supporter_tag.user_search.searching')
+      return @setUserInteraction(false)
+
+    [avatarUrl, text] = if @user
+                          [@user.avatarUrl, '']
+                        else
+                          ['', Lang.get("supporter_tag.user_search.not_found")]
+
+    @inputFeedback.textContent = text
+    # Avoid setting value to undefined
+    @el.querySelector('input[name="item[extra_data][target_id]"').value = @user?.userId ? null
+    $(@el.querySelectorAll('.js-avatar')).css
+      'background-image': "url(#{avatarUrl})"
+
+    @setUserInteraction(@user?)
+
+  updateCostDisplay: =>
+    @el.querySelector('input[name="item[cost]"').value = @cost.price()
+    @el.querySelector('input[name="item[extra_data][duration]"').value = @cost.duration()
+    @priceElement.textContent = "USD #{@cost.price()}"
+    @durationElement.textContent = @cost.durationText()
+    @discountElement.textContent = @cost.discountText()
+    @updateSliderPreset(elem, @cost) for elem in @sliderPresets
+
+  updateSliderPreset: (elem, cost) ->
+    $(elem).toggleClass('store-slider__preset--active', cost.duration() >= +elem.dataset.months)
```

Comment:
I'd say more like it's a state change with no explicit style change. It just happens the preset element style changes when it's got `js-` class attached.